lib/blocs_and_cubits/drawer_cubit.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:bloc/bloc.dart';
+
+enum DrawerState {
+  initial,
+  login,
+  register,
+}
+
+class DrawerCubit extends Cubit<DrawerState> {
+  DrawerCubit() : super(DrawerState.initial);
+  void changeToLoginState() => emit(DrawerState.login);
+  void changeToInitialState() => emit(DrawerState.initial);
+  void changeToRegisterState() => emit(DrawerState.register);
```

Comment:
```suggestion
  void changeDrawerState(DrawerState state) => emit(state);
```
Maybe it will be more clear? We're using 1 func instead of three.

Reply:
I had a brief talk with Kamil about similar part of code, and he recommended me to make separate functions for each state. However it was a bit different scenario and I don't remember the conversation that good, so maybe in this case your solution would be better. Personally I also prefer your method C:

Reply:
Haha, I've talked with Andrii about that. He prefers also 'my' method 😄